java/code/src/com/redhat/rhn/common/util/FileUtils.java

```diff
@@ -61,12 +61,13 @@ public static void writeStringToFile(String contents, String path) {
                 file.delete();
             }
             file.createNewFile();
-            Writer output = new BufferedWriter(new FileWriter(file));
-            try {
-              output.write(contents);
-            }
-            finally {
-              output.close();
+            try (FileOutputStream fos = new FileOutputStream(file);
+                 FileWriter fw = new FileWriter(fos.getFD());
+                 BufferedWriter output = new BufferedWriter(fw)) {
+                output.write(contents);
+                output.flush();
+                fw.flush();
+                fos.getFD().sync();
```

Comment:
This looks not like standard java . `try` used like `if` or I am not deep enough inside of what you can do with java.
I think we need another expert to look into this.

Reply:
@mcalmer I also didn't know about it.

This is called `try-with-resources` statement: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

It does simply close the declared resources once the statement is done.

Reply:
you can learn every day something new :-)